.github/workflows/aws_demo_runner.yml

```diff
@@ -0,0 +1,161 @@
+name: Localhost Server Demo Runner
+
+on:
+  push:
+    branches: ["**"]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  build-and-run-localhost-demos:
+    name: Build and Run Localhost Server Demos
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Clone This Repo
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Install Python3
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+
+      - name: Install libmosquitto
+        run: sudo apt-get install -y libmosquitto-dev
+
+      - name: Install CMake
+        run: |
+          curl https://cmake.org/files/v3.2/cmake-3.2.0-Linux-x86_64.tar.gz -o cmake.tar.gz
+          tar -xf cmake.tar.gz
+
+      - name: Generate SSL credentials
+        id: generate-credentials
+        uses: jasonpcarroll/CI-CD-GitHub-Actions/ssl-credential-creator@main
+
+      - name: Start localhost MQTT broker
+        id: mqtt-broker
+        uses: jasonpcarroll/CI-CD-GitHub-Actions/localhost-mqtt-broker@main
+        with:
+          root-ca-cert-path: ${{ steps.generate-credentials.outputs.root-ca-cert-path }}
+          server-priv-key-path: ${{ steps.generate-credentials.outputs.server-priv-key-path }}
+          server-cert-path: ${{ steps.generate-credentials.outputs.server-cert-path }}
+
+      - name: Start localhost HTTP server
+        id: http-server
+        uses: Eun/http-server-action@v1.0.6
+
+      - name: Configure CMake Build for Localhost servers
+        run: |
+          mkdir build && cd build
+          ../cmake-3.2.0-Linux-x86_64/bin/cmake .. \
+          -G "Unix Makefiles" \
+          -DBUILD_DEMOS=1 \
+          -DCMAKE_BUILD_TYPE=Debug \
+          -DCMAKE_C_FLAGS='-Wall -Wextra -Werror' \
+          -DAWS_IOT_ENDPOINT="localhost" \
+          -DBROKER_ENDPOINT="localhost" \
+          -DCLIENT_CERT_PATH=${{ steps.generate-credentials.outputs.device-cert-path }} \
+          -DROOT_CA_CERT_PATH=${{ steps.generate-credentials.outputs.root-ca-cert-path }} \
+          -DCLIENT_PRIVATE_KEY_PATH=${{ steps.generate-credentials.outputs.device-priv-key-path }} \
+          -DCLIENT_IDENTIFIER="test" \
+          -DTHING_NAME="thing-name" \
+          -DS3_PRESIGNED_GET_URL="get-url" \
+          -DS3_PRESIGNED_PUT_URL="put-url" \
+          -DCLAIM_CERT_PATH="cert/path" \
+          -DCLAIM_PRIVATE_KEY_PATH="key/path" \
+          -DPROVISIONING_TEMPLATE_NAME="template-name" \
+          -DDEVICE_SERIAL_NUMBER="00000" \
+          -DCSR_SUBJECT_NAME="CN=Fleet Provisioning Demo" \
+          -DGREENGRASS_ADDRESS="greengrass-address"
+
+      - name: Configure the HTTP Plaintext Demo
+        working-directory: demos/http/http_demo_plaintext
+        run: sed -i '1s/^/#define HTTP_PORT 8080\n/' demo_config.h
```

Comment:
Kind of a nit, but maybe a sed command more like `sed -i 's/define HTTP_PORT[[:blank:]]\{1,\}( 80 )/define HTTP_PORT ( 8080 )/g' demo_config.h` that way there aren't two values being defined for if searching through logs?

Reply:
Done and I updated all other sed commands as well.